### PR TITLE
Fix deletion of Airflow source dir in virtualenv installation script

### PR DESCRIPTION
**Pull request by renzepost**
---
I was following the contributors quick start guide and my source directory kept getting deleted when running this command:
`./scripts/tools/initialize_virtualenv.py`
My source directory happened to be `$HOME/airflow`. With this fix, at least the check if `airflow_home_dir == airflow_sources` works again.
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: scripts/tools/initialize_virtualenv.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def main():
     Setup local virtual environment.
     """
     airflow_home_dir = Path(os.environ.get("AIRFLOW_HOME", Path.home() / "airflow"))
-    airflow_sources = Path(__file__).parents[2]
+    airflow_sources = Path(__file__).parents[2].absolute()
```
Comment (uranusjr) [inline scripts/tools/initialize_virtualenv.py]: Do not use `absolute`, it is undocumented and known to be erroneous in edge cases.

Comment (renzepost) [reply scripts/tools/initialize_virtualenv.py]: Thanks! I've replaced it with `resolve`

**Review by uranusjr — APPROVED** [review]

File: scripts/tools/initialize_virtualenv.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def main():
     Setup local virtual environment.
     """
     airflow_home_dir = Path(os.environ.get("AIRFLOW_HOME", Path.home() / "airflow"))
-    airflow_sources = Path(__file__).parents[2]
+    airflow_sources = Path(__file__).resolve().parents[2]
```
Comment (uranusjr) [inline scripts/tools/initialize_virtualenv.py:115]: I was thinking if `airflow_sources = Path(__file__).parents[2].resolve()` is better, but it probably does not matter either way sicne we control the `scripts/tools` directory and there’s very unlikely any non-regular entries (e.g. symlinks).

**Review by amoghrajesh — APPROVED** [review]
LGTM +1

**Review by hussein-awala — APPROVED** [review]

**Comment by hussein-awala** [comment]
The failed tests are not related to this PR and they will be fixed by #31664

**Pull request merged**